server/src/main/scala/de/frosner/broccoli/controllers/WebSocketController.scala

```diff
@@ -41,45 +35,31 @@ case class WebSocketController @Inject()(webSocketService: WebSocketService,
       Logger.info(s"New connection $connectionLogString")
 
       // TODO receive string and try json decoding here because I can handle the error better
-      val in = Iteratee
-        .foreach[Msg] { incomingMessage =>
-          val outgoingMessage = Json
-            .fromJson[IncomingWsMessage](incomingMessage)
-            .map {
-              case IncomingWsMessage.AddInstance(instanceCreation) =>
-                InstanceController
-                  .create(instanceCreation, user, instanceService)
-                  .fold(AddInstanceError, AddInstanceSuccess)
-              case IncomingWsMessage.DeleteInstance(instanceId) =>
-                InstanceController
-                  .delete(instanceId, user, instanceService)
-                  .fold(DeleteInstanceError, DeleteInstanceSuccess)
-              case IncomingWsMessage.UpdateInstance(instanceUpdate: InstanceUpdate) =>
-                InstanceController
-                  .update(instanceUpdate.instanceId.get, instanceUpdate, user, instanceService)
-                  .fold(UpdateInstanceError, UpdateInstanceSuccess)
-            }
-            .recoverTotal { error =>
-              Logger.warn(s"Can't parse a message from $connectionId: $error")
-              OutgoingWsMessage.Error(s"Failed to parse message message: $error")
-            }
-          webSocketService.send(connectionId, Json.toJson(outgoingMessage))
-        }
+      val in = Enumeratee.mapM[Msg] { incomingMessage =>
+        Json
+          .fromJson[IncomingMessage](incomingMessage)
+          .map(messageHandler.processMessage(user))
+          .recoverTotal { error =>
+            Logger.warn(s"Can't parse a message from $connectionId: $error")
+            Future.successful(OutgoingMessage.Error(s"Failed to parse message message: $error"))
+          }
+      } &>> Iteratee
```

Comment:
Do you mind using the `transform` function because it is easier to google and to understand what it does from the name?